win32ss/user/ntuser/menu.c

```diff
@@ -3282,11 +3282,13 @@ static void FASTCALL MENU_HideSubPopups(PWND pWndOwner, PMENU Menu,
       {
           PWND pWnd;
           if (!VerifyMenu(Item->spSubMenu)) return;
-          pWnd = ValidateHwndNoErr(Item->spSubMenu->hWnd);
           MENU_HideSubPopups(pWndOwner, Item->spSubMenu, FALSE, wFlags);
           MENU_SelectItem(pWndOwner, Item->spSubMenu, NO_SELECTED_ITEM, SendMenuSelect, NULL);
           TRACE("M_HSP top p hm %p  pWndOwner IDMenu %p\n",top_popup_hmenu,pWndOwner->IDMenu);
-          co_UserDestroyWindow(pWnd);
+
+          pWnd = ValidateHwndNoErr(Item->spSubMenu->hWnd);
```

Comment:
The previous placement of these calls makes me think that somebody thought `Item->spSubMenu->hWnd` might change between then and now.
Both the recursive `MENU_HideSubPopups` call and the `MENU_SelectItem` call make call-outs to user mode, so whatever UM can change via an API call can change there... I'm not sure if that includes `Item->spSubMenu->hWnd`.

I'm also just generally suspicious of added null checks. Any idea WHY this is null/invalid here? If that in itself is a bug, then we shouldn't work around that here, we should fix the bug. If the window handle being null or becoming invalid is a valid state, then a comment to that effect would be really helpful. Win32k has a lot of poorly-defined state, so as we make changes we need to start defining it better, or we'll be going back and forth on these kinds of changes for years to come.

Is there a test that can trigger this condition? Can that test be modified to try to change `Item->spSubMenu->hWnd` during the above callouts? That would be really useful to avoid regressions and moving in the wrong direction.

Reply:
> Any idea WHY this is null/invalid here?

`ValidateHwndNoErr` can return NULL here in case of window creation fails. Other places in win32k always do null check before calling `co_UserDestroyWindow`.

I think the problem is: our win32k didn't handle window creation failure properly. [CORE-18821](https://jira.reactos.org/browse/CORE-18821) is the example of BSOD caused by window creation failure.